### PR TITLE
onchain-development intro-to-onchain updated

### PR DESCRIPTION
**Pull request by SAMAD101**
### Summary of Changes

- changes in title, objectives, and description
- made the summary and overview sections more elaborative and detailed
- PDA section improved

### CODE REVIEW

File: content/courses/onchain-development/intro-to-onchain.md [hunk 1]
```diff
@@ -1,44 +1,56 @@
...
+  **accounts**.
+- Solana programs are most commonly written in **Rust**, often using the
+  **Anchor** framework for simplified development.
+- Anchor generates **Interface Description Language (IDL)** files, which:
+  - Describe the program's structure and functionality
+  - Enable automatic creation of JavaScript/TypeScript client libraries
+- Solana's architecture allows for parallel execution of non-overlapping
+  transactions, contributing to its high speed and efficiency.
+- Rent is a concept in Solana where accounts must maintain a minimum balance to
+  stay alive on the blockchain.
```
Comment (mikemaccana) [inline content/courses/onchain-development/intro-to-onchain.md:27]: This is all great (no feedback, just saying).

File: content/courses/onchain-development/intro-to-onchain.md [hunk 2]
```diff
@@ -1,44 +1,56 @@
...
+makes things a lot simpler and safer. Some pros of using Anchor are:
 
-- sending incoming instructions to the right instruction handlers
-- deserializing data from incoming transactions
-- checking the accounts provided with incoming instructions, for example, to
-  check that certain accounts are of a particular type, or are distinct from
-  other accounts.
+- Security checks are implemented automatically
+- Instruction routing to appropriate handlers
+- Data serialization and deserialization from incoming transactions
```
Comment (mikemaccana) [inline content/courses/onchain-development/intro-to-onchain.md:50]:
```suggestion
- Automatic routing of incoming instructions to the correct instruction handler
- Automatic serialization and deserialization of the data inside transactions
```

File: content/courses/onchain-development/intro-to-onchain.md [hunk 3]
```diff
@@ -67,7 +81,8 @@ execute.
 
 Instruction handlers are how blockchain programs process the instructions from
 clients. Every exchange, lending protocol, escrow, oracle, etc. provides their
-functionality by instruction handlers.
+functionality by instruction handlers. They enable program to process complex
+financial operations on Solana.
```
Comment (mikemaccana) [inline content/courses/onchain-development/intro-to-onchain.md:85]: Not sure re: "They enable program to process complex financial operations on Solana." - people already know this based on the previous sentence.

File: content/courses/onchain-development/intro-to-onchain.md [hunk 4]
```diff
@@ -84,35 +99,63 @@ Programs on Solana can transfer tokens, which end up in user wallet addresses
 But more importantly, programs on Solana can create additional addresses as
 needed, to store items of data.
 
-### Programs store data in Program Derived Addresses (PDAs), a key-value store
+This approach allows Solana programs to maintain complex states and interact
+with various types of data, all within the blockchain's account structure.
```
Comment (mikemaccana) [inline content/courses/onchain-development/intro-to-onchain.md:103]:
```suggestion
This is how Solana programs store their state.
```

Try and keep it plain.

File: content/courses/onchain-development/intro-to-onchain.md [hunk 5]
```diff
@@ -84,35 +99,63 @@ Programs on Solana can transfer tokens, which end up in user wallet addresses
...
-    like `'config'`. Your program's PDAs are unique, so they won't conflict with
-    other programs.
-- The value is the data inside the account at the given address.
-  - The data inside the PDA is determined by you, the programmer.
-
-Key value stores allow your onchain program, and client software, to
-consistently determine the address for a data item because the same seeds will
-always return the same address.
+Solana's PDAs can be thought of as a **key/value store**. A PDA can be designed
+to store any form of data as required by the program.
```
Comment (mikemaccana) [inline content/courses/onchain-development/intro-to-onchain.md:109]: This is good (no feedback just saying).

File: content/courses/onchain-development/intro-to-onchain.md [hunk 6]
```diff
@@ -84,35 +99,63 @@ Programs on Solana can transfer tokens, which end up in user wallet addresses
...
+     the address of the data.
+
+3. **Data Storage**
+
+   - Programmers define the structure of data stored in PDAs
+   - Can store any type of program-specific information
+
+4. **Some properties**:
+   - PDAs are off the ED25519 elliptic curve so they are not `KeyPairs` like
+     traditional accounts on Solana, they only have a `PublicKey`.
```
Comment (mikemaccana) [inline content/courses/onchain-development/intro-to-onchain.md:133]:
```suggestion
   - PDAs are off the Ed25519 elliptic curve. While the data type web3.js uses is a `PublicKey`, PDA addresses are not public keys and do not have a matching private key. 
```

**Review by mikemaccana — CHANGES_REQUESTED** [review]
The new PDA content is ✨excellent✨ and much clearer than the previous content. A few small changes needed here but not many.

**Review by mikemaccana — CHANGES_REQUESTED** [review]
You new PDA section is ✨excellent✨ and much clearer than the previous content. A few small changes

File: content/courses/onchain-development/intro-to-onchain.md (hunk 1 above)
Comment (SAMAD101) [reply content/courses/onchain-development/intro-to-onchain.md:27]: thanks

File: content/courses/onchain-development/intro-to-onchain.md [hunk 7]
```diff
@@ -1,44 +1,56 @@
...
-- **Onchain programs** are programs that run on Solana.
-- Programs have one or more **instruction handlers**. Instruction handlers are
-  functions that process the **instructions** in Solana transactions.
-- Instruction handlers write their data to Solana **accounts** and then either
-  succeed or fail.
-- Solana programs are most commonly written in **Rust**, typically with the
-  **Anchor** framework.
-- Anchor programs have IDLs, which are small JSON files describing programs.
-  IDLs can be used to automatically create JS/TS libraries for a program.
+- **Onchain programs** refers to the code that run directly on Solana.
```
[the remark this answers is not in the document]
Comment (SAMAD101) [reply content/courses/onchain-development/intro-to-onchain.md:14]: it has a good sound to it. the analogy makes it a bit easier to imagine programs

File: content/courses/onchain-development/intro-to-onchain.md (hunk 3 above)
Comment (SAMAD101) [reply content/courses/onchain-development/intro-to-onchain.md:85]: y; that might be a bit redundant

**Comment by SAMAD101** [comment]
> The new PDA content is ✨excellent✨ and much clearer than the previous content. A few small changes needed here but not many.

Can you tell me what changes are needed?

File: content/courses/onchain-development/intro-to-onchain.md [hunk 8]
```diff
@@ -1,44 +1,57 @@
...
-- **Onchain programs** are programs that run on Solana.
-- Programs have one or more **instruction handlers**. Instruction handlers are
-  functions that process the **instructions** in Solana transactions.
-- Instruction handlers write their data to Solana **accounts** and then either
-  succeed or fail.
-- Solana programs are most commonly written in **Rust**, typically with the
-  **Anchor** framework.
-- Anchor programs have IDLs, which are small JSON files describing programs.
-  IDLs can be used to automatically create JS/TS libraries for a program.
+- **Onchain programs** run directly on Solana, just like programs on your
```
Comment (mikemaccana) [inline content/courses/onchain-development/intro-to-onchain.md:14]:
```suggestion
- **Onchain programs** (sometimes called 'smart contracts') run directly on Solana, just like programs on your
```

Comment (mikemaccana) [reply content/courses/onchain-development/intro-to-onchain.md:14]: We recently started to use 'smart contract' just for SEO and recognizability purposes.

**Comment by mikemaccana** [comment]
I just meant the changes in the comments above. They're all resolved now - thanks!

**Review by mikemaccana — APPROVED** [review]

**Comment by mikemaccana** [comment]
Approved and marked as winner

**Pull request merged**